src/app.ts

```diff
@@ -13,14 +14,73 @@ interface TemperatureSummary {
   average: number
 }
 
+/* Object containing a dictionary of dictionaries */
+const sumarys: { [key: string]: { [item: string]: TemperatureSummary } } = {}
+
+/* Function that given the array of TemperatureReadings fills the sumarys object,
+  the first dictionaries with cities and the dictionary inside of the cities
+  with the dates containing the sumary of that day */
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (let read of readings) {
+    let exist = true
+    if (read.city in sumarys) {
+      if (!(read.time.toDateString() in sumarys[read.city])) {
+        exist = false
+      }
+    } else {
+      exist = false
+    }
+    if (!exist) {
+      const cityReadings = readings.filter(
+        (item) =>
+          item.city === read.city &&
+          item.time.toDateString() === read.time.toDateString(),
+      )
+      let cityFirst = cityReadings[0]
+      let cityLast = cityReadings[cityReadings.length - 1]
+      let cityHigh = cityReadings.reduce((a, b) =>
+        a.temperature > b.temperature ? a : b,
+      )
+      let cityLow = cityReadings.reduce((a, b) =>
+        a.temperature < b.temperature ? a : b,
+      )
+      let cityAvr = 0
+      for (let item of cityReadings) {
+        cityAvr += item.temperature
+      }
+      cityAvr = cityAvr / cityReadings.length
+      if (read.city in sumarys) {
+        sumarys[read.city][read.time.toDateString()] = {
+          first: cityFirst.temperature,
+          last: cityLast.temperature,
+          high: cityHigh.temperature,
+          low: cityLow.temperature,
+          average: cityAvr,
```

Comment:
I'm going to update the challenge wording, but this average is the average for the day. Lets also add another `average` field on to the city itself with the all time average temp